WordPress/src/main/java/org/wordpress/android/util/crashlogging/WPCrashLoggingDataProvider.kt

```diff
@@ -0,0 +1,116 @@
+package org.wordpress.android.util.crashlogging
+
+import android.content.SharedPreferences
+import com.automattic.android.tracks.crashlogging.CrashLoggingDataProvider
+import com.automattic.android.tracks.crashlogging.CrashLoggingUser
+import com.automattic.android.tracks.crashlogging.EventLevel
+import com.automattic.android.tracks.crashlogging.ExtraKnownKey
+import org.wordpress.android.BuildConfig
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.store.AccountStore
+import org.wordpress.android.util.BuildConfigWrapper
+import org.wordpress.android.util.EncryptedLogging
+import org.wordpress.android.util.LocaleManagerWrapper
+import org.wordpress.android.util.LogFileProviderWrapper
+import org.wordpress.android.viewmodel.ResourceProvider
+import java.util.Locale
+import javax.inject.Inject
+
+class WPCrashLoggingDataProvider @Inject constructor(
+    private val sharedPreferences: SharedPreferences,
+    private val resourceProvider: ResourceProvider,
+    private val accountStore: AccountStore,
+    private val localeManager: LocaleManagerWrapper,
+    private val encryptedLogging: EncryptedLogging,
+    private val logFileProvider: LogFileProviderWrapper,
+    private val buildConfig: BuildConfigWrapper
+) : CrashLoggingDataProvider {
+
+    override val buildType: String = BuildConfig.BUILD_TYPE
+    override val enableCrashLoggingLogs: Boolean = BuildConfig.DEBUG
+    override val locale: Locale
+        get() = localeManager.getLocale()
+    override val releaseName: String = BuildConfig.VERSION_NAME
+    override val sentryDSN: String = BuildConfig.SENTRY_DSN
+
+    override fun applicationContextProvider(): Map<String, String> {
+        return emptyMap()
+    }
+
+    override fun crashLoggingEnabled(): Boolean {
+        if (buildConfig.isDebug()) {
```

Comment:
I use `BuildConfigWrapper` here as this is something I mock in tests. In other places, static `BuildConfig` fields are used. This results in some inconsistency but on the other hand, I'd have to add getters for all used fields in `BuildConfigWrapper` - what do you prefer?